### ISSUE TITLE
Mixed scripts above target trigger error when mixed=false

### ISSUE THREAD

**Issue by facewindu** [flyway/flyway#2392]
###### Which version and edition of Flyway are you using?

5.2.4 Community Edition

###### If this is not the latest version, can you reproduce the issue with the latest one as well?
yes

###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin)
Java API

###### Which database are you using (type & version)?
PG 11

###### Which operating system are you using?
macOS

###### What did you do?
I have `mixed = false`, meaning I cannot have transactional and non-transactional statements in the same migration script.
I have a unit test with 3 SQL migration scripts
V0: with transactional statements
V1: with non-transactional statements
V2: with mixed transactional and non-transactional statements

I have a unit test that tests that I can perform a migration from scratch to version 1, even if I have a 'pending' V2 script that would fail with `mixed=false`.
This test was working with Flyway 5.1.4, because when resolving the migrations, only the migration script path was considered at that point (in this [code block](https://github.com/flyway/flyway/blob/flyway-5.1.4/flyway-core/src/main/java/org/flywaydb/core/internal/resolver/sql/SqlMigrationResolver.java#L147-L159))

In Flyway 5.2.4, a SqlScript object is created from all found migrations, and this fail when parsing the V2 script  (see [this](https://github.com/flyway/flyway/blob/flyway-5.2.4/flyway-core/src/main/java/org/flywaydb/core/internal/resolver/sql/SqlMigrationResolver.java#L140) and then [this](https://github.com/flyway/flyway/blob/flyway-5.2.4/flyway-core/src/main/java/org/flywaydb/core/internal/sqlscript/SqlScript.java#L161))

Basically, this check is done 'too soon', as all migrations scripts that are found are checked, even though some are 'pending', and this is ok to let them break the 'non-mixed' rule as long as they are not applied.

Is there a way for me to go back to the old behaviour, that allows to have scripts breaking the rule, as long as they are not applied when performing a migration ?

###### What did you expect to see?

I would have expected the same behaviour. Let the V2 script be, since it's not intended to be applied (target version = 1), and not fail.

###### What did you see instead?

Exception thrown due to V2 breaking the 'non-mixed statements' rule.

**Comment by axelfontaine** [comment]
I can definitely see where you are coming from. We'll think about it for 6.0.
